src/tracker-component.js

```diff
@@ -98,24 +115,19 @@ const track = <T>(config : Config, trackingType : TrackingType, trackingData : T
 const trackCartEvent = <T>(config : Config, cartEventType : CartEventType, trackingData : T) =>
     track(config, 'cartEvent', { ...trackingData, cartEventType });
 
-const generateId = () : string =>
-    Math.random()
-        .toString(16)
-        .slice(2);
-
-export const Tracker = (config? : Config = { user: { id: generateId() } }) => ({
+export const Tracker = (config? : Config = { user: { email: undefined, name: undefined } }) => ({
     view:           (data : ViewData) => track(config, 'view', data),
     addToCart:      (data : CartData) => trackCartEvent(config, 'addToCart', data),
     setCart:        (data : CartData) => trackCartEvent(config, 'setCart', data),
     removeFromCart: (data : RemoveCartData) => trackCartEvent(config, 'removeFromCart', data),
     purchase:       (data : PurchaseData) => track(config, 'purchase', data),
     setUser:        (data : UserData) => {
-        const oldUserId = config.user ? config.user.id : undefined;
-        config.user = config.user || { id: data.user.id };
-        config.user.id = data.user.id || config.user.id;
-        config.user.email = data.user.email || config.user.email;
-        config.user.name = data.user.name || config.user.name;
-        track(config, 'setUser', { oldUserId });
+        config.user = {
```

Comment:
Mutating config feels weird here -- we're changing an object that we didn't create. Would it make sense to clone this object before we mutate it?

Reply:
do you mean changing it so it's like this instead?
```
config = {
    ...config,
    user: {
        ...config.user,
        email: data.user.email || ((config && config.user) || {}).email,
        name:  data.user.name || ((config && config.user) || {}).name
    }
};
```

Reply:
👍